aips/aip-x.md

```diff
@@ -0,0 +1,707 @@
+
+The following fields MUST be verified to prevent phishing attacks: `domain` , `address` , `uri` , `version` , and `chainId` . If the fields do not match the signing request, errors must be displayed to the users detailing them out. It is suggested to add additional steps to the approval process, such as an opt-in checkmark field, that confirms the user’s intent if errors are present.
+
+A wallet may also decide to prevent the approval of invalid signing requests.
+
+## Reference Implementation
+
+### Interfaces
+
+#### @aptos-labs/siwa
+
+A new package named `@aptos-labs/siwa` will be created to house the utilities for the _Sign in with Aptos_ standard. These utilities will be responsible for creating messages, verifying messages, and deserializing public key and/or signatures according.
+
+Below are the primary functions that that will be provided as part of the `@aptos-labs/siwa` utilities:
+
+```ts
+/**
+ * Create a SignIn message text from the input following the ABNF format defined in the Sign in with Aptos
+ * specifications.
+ *
+ * @param input The input to create the SignIn message text from.
+ *
+ * @returns The SignIn message text.
+ */
+function createSignInMessageText(input: AptosSignInMessage["fields"]): string;
+
+/**
+ * Verifies an input SignIn message against expected fields.
+ *
+ * @param input The input to verify the message against.
+ * @param expected The expected message to verify against the input.
+ *
+ * @returns The verification result.
+ */
+function verifySignInMessage(
+  input: AptosSignInInput,
+  expected: AptosSignInMessage | AptosSignInMessage["fields"] | string
+): VerificationResult<AptosSignInMessage["fields"]>;
+
+/**
+ * Verifies outputs from a `signIn` method response against input fields.
+ *
+ * @param input The input to verify the output against.
+ * @param output The output to verify against the input.
+ *
+ * @returns The verification result.
+ */
+function verifySignIn(
+  input: AptosSignInInput,
+  output: {
+    publicKey: PublicKey;
+    signature: Signature;
+    message: string;
+  }
+): VerificationResult<AptosSignInMessage["fields"]>;
+
+/**
+ * Generate a signing message using the Sign in with Aptos signing algorithm.
+ * sha3_256( sha3_256(b"SIGN_IN_WITH_APTOS::" ) || <message> )
+ *
+ * @param message The SIWA message to sign.
+ 
+ * @returns The signing message.
+ */
+function generateSignInSigningMessage(message: string): Uint8Array;
+```
+
+#### @aptos-labs/wallet-standard
+
+The following interfaces would be introduced into the `@aptos-labs/wallet-standard` package.
+
+```ts
+import { AccountInfo, Signature } from "@aptos-labs/wallet-standard";
+
+export type AptosSignInInput = {
+  domain?: string;
+  address?: string;
+  uri?: string;
+  version?: string;
+  statement?: string;
+  nonce?: string;
+  chainId?: string;
+  issuedAt?: string;
+  expirationTime?: string;
+  notBefore?: string;
+  requestId?: string;
+  resources?: string[];
+};
+
+export type AptosSignInMessage = {
```

Comment:
I think you should solely use this `AptosSignInMessage` term throughout the text, instead of synonyms for it.